### PR TITLE
fix(941390): add missing javascript `prompt` and `confirm` methods

### PR DESCRIPTION
**Pull request by Xhoenix**
Add missing Javascript interaction methods: `prompt()` and `confirm()`.

Fixes 9KH-231120-1 and #3381.

Reference: https://javascript.info/alert-prompt-confirm

### CODE REVIEW

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941390.yaml [hunk 1]
```diff
@@ -117,3 +117,19 @@ tests:
...
             log_contains: id "941390"
+  - test_title: 941390-8
+    desc: "JavaScript call prompt"
+    stages:
+      - stage:
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              Accept: "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.>
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941390.yaml:128]:
```suggestion
              Accept: "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.>"
```

**Review by fzipi — CHANGES_REQUESTED** [review]
Thanks for your PR.

**Comment by dune73** [comment]
PR looks good to me with the exception of the missing double quote. Happy to merge afterwards.

**Review by dune73 — APPROVED** [review]

**Comment by fzipi** [comment]
@Xhoenix Can you please add also `confirm`? And maybe adding a new test for it. Also, maybe adding a new reference to https://javascript.info/alert-prompt-confirm.

**Comment by fzipi** [comment]
@RedXanadu Is there a way to test these two word additions to the rule using your script? I'm afraid adding them will be way more FP prone 🤔

**Review by theMiddleBlue — APPROVED** [review]
LGTM

**Comment by RedXanadu** [comment]
I'll run before and after numbers and compare…

**Comment by RedXanadu** [comment]
@Xhoenix A question about the change to `rules/unix-shell.data`: is that a commit that belongs to a different issue/PR? I believe that file relates to the 93216x rules.

**Comment by RedXanadu** [comment]
The new regular expression does not cause any false positives with the set of 10,000 English test sentences. I even tried the corpus of 1,000,000 sentences just to be safe, and again there were 0 false positives caused by this new regex.

Inevitably, there _will_ be natural language false positives with this eventually. It's easy to create some, e.g.:

* Be prompt (to every meeting!)
* I can confirm (with regret) that the coffee machine is broken.

But it's easy to work backwards from a regex and create false positive payloads… Certainly, nothing triggered in the news article test sentences :slightly_smiling_face:

**Review by RedXanadu — CHANGES_REQUESTED** [review]
@Xhoenix A question about the change to `rules/unix-shell.data`: is that a commit that fixes / belongs to a different issue/PR? I believe that file relates to the 93216x rules.

**Comment by dune73** [comment]
That `pyversions` and `py3versions` had me puzzled too. I suggest we keep PRs separate.

So the idea would be to remove that change from this change and open a separate PR for it. It's a welcome addition to unix-shell.data.

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941390.yaml [hunk 2]
```diff
@@ -117,3 +117,35 @@ tests:
...
+      - stage:
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              Accept: "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.>"
+              User-Agent: "OWASP CRS test agent"
+            method: GET
+            port: 80
+            uri: '/?arg=prompt%28%29'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941390.yaml:132]:
```suggestion
            uri: '/get?arg=prompt%28%29'
```

File: tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941390.yaml [hunk 3]
```diff
@@ -117,3 +117,35 @@ tests:
...
+      - stage:
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              Accept: "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.>"
+              User-Agent: "OWASP CRS test agent"
+            method: GET
+            port: 80
+            uri: '/?arg=confirm%28%29'
```
Comment (fzipi) [inline tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941390.yaml:148]:
```suggestion
            uri: '/get?arg=confirm%28%29'
```

**Review by fzipi — COMMENTED** [review]
One last small change, so we keep tests clean.

**Review by RedXanadu — APPROVED** [review]

**Comment by dune73** [comment]
Cool. Let's get the HTTP methods into the URI and we're ready to merge.

**Review by fzipi — APPROVED** [review]

**Pull request merged**